sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/CountIf.scala

```diff
@@ -0,0 +1,73 @@
+/*
```

Comment:
instead of creating a new expression, can't we just rewrite it in the analyzer/optimizer?

Reply:
I think if we are going to add this function, rewrite it in analyzer would be better. As @HyukjinKwon suggested, we can also add sum_if, avg_if... And they will be easy to rewrite in analyzer, where the same logic can be easily reused.

Reply:
I didn't suggest but I asked if there are multiple stuff like `sum_of`, `avg_if`. 

It's a different to add one function vs a set of multiple functions. If it needs a bunch of huge codes to add those multiple functions, it's a different story, than adding a single function to match.